src/pyhf/readxml.py

```diff
@@ -46,15 +47,15 @@ def import_root_histogram(rootdir, filename, path, name, filecache=None):
     else:
         f = filecache[fullpath]
     try:
-        h = f[name]
-    except KeyError:
+        hist = f[name]
+    except (KeyError, uproot.deserialization.DeserializationError):
         try:
-            h = f[str(Path(path).joinpath(name))]
+            hist = f[str(Path(path).joinpath(name))]
```

Comment:
I'm not sure this is the recommended way to do it? Do we guarantee that uproot supports these path semantics correctly (e.g. on windows)?

Reply:
Uproot supports Windows pathnames by passing them to `os.path` routines. There are tests on Windows, and the `skhep_test_datapath` probably gives an absolute Windows pathname and thus it would have been tested, though I haven't checked.

Uproot also supports the new `Path` objects, but it does so by converting them into strings once you've passed them to `uproot.open`. Therefore, you can do all the `Path` gymnastics within the `uproot.open` argument list. Passing a `Path` object to `uproot.open` is also a way to avoid issues with colon (`:`) in the filename being interpreted as a separator between filesystem paths and within-ROOT file paths.

However, paths _within_ a ROOT file are not filesystem paths and `Path` is not accepted in `ReadOnlyDirectory.__getitem__`. Using `Path.joinpath` is probably a bad idea because it might get joined by backslashes on Windows, but ROOT (and Uproot) never accepts backslashes as a TDirectory separator, even on Windows. Also, there are semicolons (`;`) for cycle numbers and all that. These paths, which are actually Mapping keys, should be strings and should be built with string-manipulation, such as `"/".join`.

Reply:
How do you handle the case when say `"/".join([path, name])` has a trailing slash in `path` ? Does `uproot` handle `//` ok or is this something we'd need to check and handle properly? In which case, we'd probably need a custom `joinpath` utility in `pyhf` just to handle this case, but then I could argue this might be a quick utility that `uproot` could have instead... but I'd rather not give you more work.

Reply:
A quick check: Uproot handles stray slashes.

```python
>>> import skhep_testdata
>>> import uproot
>>> f = uproot.open(skhep_testdata.data_path("uproot-hepdata-example.root"))
>>> f["hpx"]
<TH1F (version 1) at 0x7f209a3cee50>
>>> f["/hpx"]
<TH1F (version 1) at 0x7f209a3cee50>
>>> f["hpx/"]
<TH1F (version 1) at 0x7f209a3cee50>
>>> f["/hpx/"]
<TH1F (version 1) at 0x7f209a3cee50>
>>> f["//hpx//"]
<TH1F (version 1) at 0x7f209a3cee50>

>>> f = uproot.open(skhep_testdata.data_path("uproot-nesteddirs.root"))
>>> f["one/two/tree"]
<TTree 'tree' (20 branches) at 0x7f209a1c3730>
>>> f["/one/two/tree/"]
<TTree 'tree' (20 branches) at 0x7f209a1c3730>
>>> f["//one//two//tree//"]
<TTree 'tree' (20 branches) at 0x7f209a1c3730>
```